dlog/plonk-5-wires/src/index.rs

```diff
@@ -0,0 +1,169 @@
+/*****************************************************************************************************************
+
+This source file implements Plonk Protocol Index primitive.
+
+*****************************************************************************************************************/
+
+use ff_fft::{DensePolynomial, Radix2EvaluationDomain as D};
+use commitment_dlog::{srs::SRS, CommitmentField, commitment::{CommitmentCurve, PolyComm}};
+use oracle::poseidon_5_wires::PlonkSpongeConstants;
+use oracle::poseidon::{ArithmeticSpongeParams, SpongeConstants};
+use plonk_5_wires_circuits::{constraints::{zk_w, ConstraintSystem}, wires::COLUMNS};
+use array_init::array_init;
+use algebra::AffineCurve;
+use algebra::PrimeField;
+
+type Fr<G> = <G as AffineCurve>::ScalarField;
+type Fq<G> = <G as AffineCurve>::BaseField;
+
+pub enum SRSValue<'a, G : CommitmentCurve> {
```

Comment:
This (and the related functions) might be better in `commitment_dlog`, since they're common to all of the dlog proof systems.